src/screens/SignUp/PersonalInfo.tsx

```diff
@@ -99,18 +99,108 @@ function PersonalInfoScreen() {
       return false;
     }
 
-    if (!/^\(\d{3}\)\s\d{3}-\d{4}/.test(secondContactNumber)) {
+    if (!/^\d{10}$/.test(secondContactNumber)) {
       return false;
     }
 
-    return true;
+    // Following check if the date matches number of days in a month
+    const [monthString, dateString, yearString] = dateofBirth.split("-");
+    const [month, date, year] = [
+      parseInt(monthString, 10),
+      parseInt(dateString, 10),
+      parseInt(yearString, 10),
+    ];
+
+    if (month > 12 || date > 31 || year < 1900) {
+      return false;
+    }
+
+    if (month === 2) {
+      const isLeapYear =
+        (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0;
+      if (isLeapYear && date <= 29) {
+        return true;
+      }
+      if (!isLeapYear && date <= 28) {
+        return true;
+      }
+      return false;
+    }
+
+    if (month < 8) {
+      if (month % 2 === 0) {
+        return date <= 30;
+      }
+      return date <= 31;
+    }
+    if (month % 2 === 0) {
+      return date <= 31;
+    }
+    return date <= 30;
+  };
+
+  const formatPhoneNumber = (currentNumber: string) => {
+    const digitsOnly = currentNumber.replace(/\D/g, "");
+
+    if (digitsOnly.length < 3) {
+      return digitsOnly;
+    }
+    if (digitsOnly.length < 6) {
+      return `(${digitsOnly.slice(0, 3)})${digitsOnly.slice(3)}`;
+    }
+    return `(${digitsOnly.slice(0, 3)})${digitsOnly.slice(
```

Comment:
Add a space after the closed parenthesis of the phone number (eg: "(123) 234-1234" instead of "(123)234-1234")